### PR TITLE
Network ipv6 enable

### PR DESCRIPTION
**Pull request by keszybz**
/cc @yuwata, @ssahani

### CODE REVIEW

**Comment by yuwata** [comment]
The original motivation of #13283 is always making ipv6 enabled if any ipv6 setting is enabled. Moreover, the cached value must be 'updated' before any call of `link_sysctl_ipv6_enabled()`. Otherwise, e.g., dhcpv6 may be disabled by the sysctl value.

**Comment by ssahani** [comment]
LGTM

**Comment by keszybz** [comment]
Updated again.

> The original motivation of #13283 is always making ipv6 enabled if any ipv6 setting is enabled. 

Yes. Now the second patch simply drops all checks of the sysctl. We just write the sysctl value if appropriate. PTAL.

File: src/network/networkd-link.c [hunk 1]
```diff
@@ -340,9 +307,11 @@ static int link_enable_ipv6(Link *link) {
...
                 r = sysctl_write_ip_property_boolean(AF_INET6, link->ifname, "disable_ipv6", false);
                 if (r < 0)
-                        log_link_warning_errno(link, r, "Cannot enable IPv6: %m");
-                else
-                        log_link_info(link, "IPv6 successfully enabled");
+                        return log_link_warning_errno(link, r, "Cannot enable IPv6: %m");
+
+                log_link_info(link, "IPv6 successfully enabled");
+                /* We know what the sysctl value is, so update our internal cache. */
+                link->sysctl_ipv6_enabled = true;
```
Comment (yuwata) [inline src/network/networkd-link.c]: OK, then please drop `Link::sysctl_ipv6_enabled`. It is not used anymore.

Comment (keszybz) [reply src/network/networkd-link.c]: Thanks. Updated.

**Comment by yuwata** [comment]
LGTM.

**Comment by keszybz** [comment]
CentOS CI (Arch in KVM) fails with `Process 23537 (qemu-system-x86) of user 0 dumped core.`.
But systemd-networkd-tests.py so that should be OK. @mrc0mmand you might want to take a look.

**Comment by mrc0mmand** [comment]
Thanks for the heads up. Interestingly enough, even though qemu segfaulted somewhere along the way, all tests finished successfully. I'm tempted to just update the [coredumpctl regex](https://github.com/systemd/systemd-centos-ci/blob/master/common/utils.sh#L180-L182) to filter out qemu segfaults as well so it doesn't bother us in the future.

**Pull request merged**